test/test-network/systemd-networkd-tests.py

```diff
@@ -5031,37 +5031,56 @@ def test_keep_configuration_on_restart(self):
-        with tempfile.TemporaryFile(mode='r+', prefix='ip_monitor') as logfile:
-            process = subprocess.Popen(['ip', 'monitor', 'dev', 'dummy98'], stdout=logfile, text=True)
-            restart_networkd()
+        with tempfile.TemporaryFile(mode='r+', prefix='ip_monitor_u') as logfile_unmanaged:
+            process_u = subprocess.Popen(['ip', 'monitor', 'dev', 'unmanaged0'], stdout=logfile_unmanaged, text=True)
+
+            copy_network_unit('12-dummy.netdev', '85-static-ipv6.network')
+            start_networkd()
             self.check_keep_configuration_on_restart()
 
-            process.send_signal(signal.SIGTERM)
-            process.wait()
+            # Start `ip monitor` with output to a temporary file
+            with tempfile.TemporaryFile(mode='r+', prefix='ip_monitor') as logfile:
+                process = subprocess.Popen(['ip', 'monitor', 'dev', 'dummy98'], stdout=logfile, text=True)
+
+                restart_networkd()
+                self.check_keep_configuration_on_restart()
+
+                process.send_signal(signal.SIGTERM)
+                process.wait()
 
-            print('### ip monitor dev dummy98 BEGIN')
+                print('### ip monitor dev dummy98 BEGIN')
+
+                # Read the `ip monitor` output looking for network changes
+                logfile.seek(0)
+                for line in logfile:
+                    print(line, end="")
+                    # Check if a link went down
+                    self.assertNotRegex(line, 'dummy98: .* state DOWN')
+                    # Check if an address was removed
+                    self.assertNotRegex(line, '^Deleted .* 2001:db8:')
+                    self.assertNotRegex(line, '^Deleted 2001:db8:.*/64')
+                    # Check if the default route was removed
+                    self.assertNotRegex(line, '^Deleted default via fe80::f0ca:cc1a')
+
+            print('### ip monitor dev dummy98 END')
+
+            process_u.send_signal(signal.SIGTERM)
+            process_u.wait()
+
+            print('### ip monitor dev unmanaged0 BEGIN')
 
             # Read the `ip monitor` output looking for network changes
-            logfile.seek(0)
-            for line in logfile:
+            logfile_unmanaged.seek(0)
+            for line in logfile_unmanaged:
                 print(line, end="")
-                # Check if a link went down
-                self.assertNotRegex(line, 'unmanaged0: .* state DOWN')
-                self.assertNotRegex(line, 'dummy98: .* state DOWN')
-                # Check if an address was removed
-                self.assertNotRegex(line, '^Deleted .* 2001:db8:')
-                self.assertNotRegex(line, '^Deleted 2001:db8:.*/64')
-                # Check if the default route was removed
-                self.assertNotRegex(line, '^Deleted default via fe80::f0ca:cc1a')
+                # Check if something happened
+                self.assertNotEmpty(line)
```

Comment:
Hmm I don't understand how this checks if something happened

Reply:
I think it is OK if it works. But, maybe better to ensure that the interface is not modified by NetworkManager by setting ID_NET_MANAGED_BY=io.systemd.Network property.

Reply:
@yuwata what's the preferred way to set the property, via a networkd .link file in `/etc/systemd/network` or a udev one in `/etc/udev/rules.d`?

Reply:
Please set through a .link file. You can find an example at e.g. /network/80-container-vb.link .